tokio/src/sync/oneshot.rs

```diff
@@ -1248,6 +1259,16 @@ impl<T> Drop for Inner<T> {
                 self.tx_task.drop_task();
             }
         }
+
+        unsafe {
+            // SAFETY: we have `&mut self`, and therefore we have
+            // exclusive access to the value.
+            //
+            // Note: the assertion holds because if the value has been sent by sender,
+            // we must ensure that the value must have been consumed by the receiver before
+            // dropping the `Inner`.
+            debug_assert!(self.consume_value().is_none());
+        }
```

Comment:
Please move safety comments before the unsafe block instead of inside it.
```suggestion
        // SAFETY: we have `&mut self`, and therefore we have
        // exclusive access to the value.
        //
        // Note: the assertion holds because if the value has been sent by sender,
        // we must ensure that the value must have been consumed by the receiver before
        // dropping the `Inner`.
        debug_assert!(unsafe { self.consume_value() }.is_none());
```
(This is not the only instance of this.)